crates/ursa-consensus/src/server.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{AbciQueryQuery, BroadcastTxQuery};
+use multiaddr::{Multiaddr, Protocol};
+use narwhal_types::{TransactionProto, TransactionsClient};
+use tendermint_proto::abci::ResponseQuery;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::oneshot::{channel as oneshot_channel, Sender as OneShotSender};
+use tracing::{debug, error};
+use warp::{Filter, Rejection, Reply};
+
+/// Simple HTTP API server which listens to messages on:
+/// * `broadcast_tx`: forwards them to Narwhal's mempool/worker socket, which will proceed to put
+/// it in the consensus process and eventually forward it to the application.
+/// * `abci_query`: forwards them over a channel to a handler (typically the application).
+pub struct AbciApi<T> {
+    mempool_address: String,
```

Comment:
```suggestion
    mempool_address: Multiaddr,
```

Reply:
Narwhals TransactionClient doesnt support Multiaddr so if we change it on the struct we would have to do the conversation every time we receive a request instead of once in the new() function